Add WCOSS2 support in configs and setup scripts

KateFriedman-NOAA opened the pull request:
- Add needed WCOSS2 support in configs and setup scripts. Tested setup scripts successfully on Cactus (have not yet tested xml).
- Created and added WCOSS2.env file. Have not yet tested at runtime.

Refs: #399

KateFriedman-NOAA committed:
Create WCOSS2 env file

Refs: #399

KateFriedman-NOAA committed:
Merge branch 'feature/ops-wcoss2' of https://github.com/KateFriedman-NOAA/global-workflow into feature/ops-wcoss2

* 'feature/ops-wcoss2' of https://github.com/KateFriedman-NOAA/global-workflow:

KateFriedman-NOAA committed:
Add WCOSS2 to configs and setup

- set default MACHINE in base configs to WCOSS2
- add WCOSS2 to fcst if-blocks for OUTPUT_FILETYPES
- add WCOSS2 to resource configs (set npe_node_max=128)
- add WCOSS2 support in setup scripts (strip out init job from ff mode)
- update upp module version to upp/8.1.0 (will remove later)

Refs: #399

aerorahul commented on env/WCOSS2.env:
```diff
@@ -0,0 +1,232 @@
+export KMP_STACKSIZE=2048M
+export KMP_AFFINITY=scatter
+export job=${LSB_JOBNAME:-$step}
+export jobid=${job}.${LSB_JOBID:-$$}
```
Are you should this is `LSB_JOBID`?  This is a PBS scheduler.

KateFriedman-NOAA replied:
Good catch, this should be a PBS variable, have now fixed, thanks!

aerorahul commented on env/WCOSS2.env:
```diff
@@ -0,0 +1,232 @@
+export MP_STDOUTMODE="ORDERED"
+export KMP_STACKSIZE=2048M
+export KMP_AFFINITY=scatter
+export job=${LSB_JOBNAME:-$step}
```
Are you should this is `LSB_JOBNAME`? This is a PBS scheduler.

KateFriedman-NOAA replied:
Fixed, thanks!

aerorahul reviewed (commented):
looks ok to me.
just a couple of comments.

KateFriedman-NOAA committed:
Update WCOSS2 env file to use PBS variables

- update job variable to use PBS_JOBNAME
- update jobid variable to use PBS_JOBID

Refs: #399

KateFriedman-NOAA commented:
@aerorahul I believe I have addressed your comments, thanks!

aerorahul commented on env/WCOSS2.env:
```diff
@@ -0,0 +1,232 @@
+export npe_node_max=128
+
+# Configure MPI environment
+export MPI_LABELIO=YES
+export MP_STDOUTMODE="ORDERED"
+export KMP_STACKSIZE=2048M
+export KMP_AFFINITY=scatter
```
This is fine for now, but we should ask the system owner if these are advantageous or not needed on this machine.

KateFriedman-NOAA replied:
Definitely...these are what is used on the Dells right now so I need inputs from others on what (if any) to set here.

aerorahul reviewed (approved):
looks good to me.

KateFriedman-NOAA commented:
Thanks @aerorahul !
